test.js

```diff
@@ -636,12 +636,97 @@ function runTests(baseopts) {
       var testPath = getFixturePath('unlink.txt');
-      watcher = chokidar.watch([testPath, testPath + '.does-not-exist'], options)
+      watcher = chokidar.watch([testPath], options)
+        .on('unlink', unlinkSpy)
+        .on('add', addSpy)
+        .on('ready', function() {
+          w(fs.unlink.bind(fs, testPath, simpleCb))();
+          waitFor([unlinkSpy], w(function() {
+            unlinkSpy.should.have.been.calledWith(testPath);
+            w(fs.writeFile.bind(fs, testPath, 're-added', simpleCb))();
+            waitFor([addSpy], function() {
+              addSpy.should.have.been.calledWith(testPath);
+              done();
+            });
+          }));
+        });
+    });
+    it('should detect unlink and re-add unfazed by watching a second file', function(done) {
+      options.ignoreInitial = true;
+      var unlinkSpy = sinon.spy(function unlinkSpy(){});
+      var addSpy = sinon.spy(function addSpy(){});
+      var testPath = getFixturePath('unlink.txt');
+      var otherPath = getFixturePath('other.txt');
+      fs.writeFileSync(otherPath, 'other');
+      watcher = chokidar.watch([testPath, otherPath], options)
+        .on('unlink', unlinkSpy)
+        .on('add', addSpy)
+        .on('ready', function() {
+          w(fs.unlink.bind(fs, testPath, simpleCb))();
+          waitFor([unlinkSpy], w(function() {
+            unlinkSpy.should.have.been.calledWith(testPath);
+            w(fs.writeFile.bind(fs, testPath, 're-added', simpleCb))();
+            waitFor([addSpy], function() {
+              addSpy.should.have.been.calledWith(testPath);
+              done();
+            });
+          }));
+        });
+    });
+    it('should detect unlink and re-add unfazed by in other directory watching a second file that does not exist', function(done) {
+      options.ignoreInitial = true;
+      var unlinkSpy = sinon.spy(function unlinkSpy(){});
+      var addSpy = sinon.spy(function addSpy(){});
+      var testPath = getFixturePath('unlink.txt');
+      var otherDirPath = getFixturePath('other-dir');
+      var otherPath = getFixturePath('other-dir/other.txt');
+      fs.mkdirSync(otherDirPath, 0x1ed);
+      // intentionally for this test don't write fs.writeFileSync(otherPath, 'other');
+      watcher = chokidar.watch([testPath, otherPath], options)
+        .on('unlink', unlinkSpy)
+        .on('add', addSpy)
+        .on('ready', function() {
+          w(fs.unlink.bind(fs, testPath, simpleCb))();
+          waitFor([unlinkSpy], w(function() {
+            unlinkSpy.should.have.been.calledWith(testPath);
+            w(fs.writeFile.bind(fs, testPath, 're-added', simpleCb))();
+            waitFor([addSpy], function() {
+              addSpy.should.have.been.calledWith(testPath);
+              done();
+            });
+          }));
+        });
+    });
+    it('should detect unlink and re-add unfazed by in same directory watching a second file that does not exist', function(done) {
```

Comment:
`should detect unlink and re-add while watching a non-existent second file in the same directory`